test/functional/cmdLineTests/modularityddrtests/modularityddrtests.xml

```diff
@@ -630,4 +631,304 @@
+		<output regex="no" type="failure">DDRInteractiveCommandException</output>
+	</test>
+
+	<test id="Run !dumpmodule with too many arguments">
+		<exec command="sh" capture="LOGNAME" platforms="zos.*" >
+			<arg>-c</arg>
+			<arg>echo $$LOGNAME</arg>
+		</exec>
+		<exec command="cp //'$LOGNAME$.J9CORE.DMP.X001' $DUMPFILE$" platforms="zos_390-64.*" />
+		<exec command="cp //'$LOGNAME$.J9CORE.DMP' $DUMPFILE$" platforms="zos_390-31.*" />
+		<exec command="tso delete J9CORE.DMP.*" platforms="zos_390-64.*" />
+		<exec command="tso delete J9CORE.DMP" platforms="zos_390-31.*" />
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core</arg>
+			<arg>$DUMPFILE$</arg>
+			<input>!dumpmodule too many options</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success">Argument failed to parse or was parsed to an unhandled subcommand.</output>
+		<output regex="no" type="required">Usage:</output>
+		<output regex="no" type="required">!dumpmodule all</output>
+		<output regex="no" type="required">!dumpmodule requires</output>
+		<output regex="no" type="required">!dumpmodule exports</output>
+		<output regex="no" type="required">!dumpmodule classes</output>
+		<output regex="no" type="required">!dumpmodule packages</output>
+		<output regex="no" type="failure">DDRInteractiveCommandException</output>
+	</test>
+
+	<test id="Run !dumpmodule with a non-numeric module address">
+		<exec command="sh" capture="LOGNAME" platforms="zos.*" >
+			<arg>-c</arg>
+			<arg>echo $$LOGNAME</arg>
+		</exec>
+		<exec command="cp //'$LOGNAME$.J9CORE.DMP.X001' $DUMPFILE$" platforms="zos_390-64.*" />
+		<exec command="cp //'$LOGNAME$.J9CORE.DMP' $DUMPFILE$" platforms="zos_390-31.*" />
+		<exec command="tso delete J9CORE.DMP.*" platforms="zos_390-64.*" />
+		<exec command="tso delete J9CORE.DMP" platforms="zos_390-31.*" />
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core</arg>
+			<arg>$DUMPFILE$</arg>
+			<input>!dumpmodule NotANumber</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success">Problem running command:</output>
+		<output regex="no" type="required">The argument "NotANumber" is not a valid number. It should be the address of a J9Module.</output>
+	</test>
+
+	<test id="Run !dumpmodule with an invalid module address">
+		<exec command="sh" capture="LOGNAME" platforms="zos.*" >
+			<arg>-c</arg>
+			<arg>echo $$LOGNAME</arg>
+		</exec>
+		<exec command="cp //'$LOGNAME$.J9CORE.DMP.X001' $DUMPFILE$" platforms="zos_390-64.*" />
+		<exec command="cp //'$LOGNAME$.J9CORE.DMP' $DUMPFILE$" platforms="zos_390-31.*" />
+		<exec command="tso delete J9CORE.DMP.*" platforms="zos_390-64.*" />
+		<exec command="tso delete J9CORE.DMP" platforms="zos_390-31.*" />
+		<command command="$JDMPVIEW_EXE$">
+			<arg>-core</arg>
+			<arg>$DUMPFILE$</arg>
+			<input>!dumpmodule 0x0</input>
+			<input>quit</input>
+		</command>
+		<output regex="no" type="success">Memory Fault reading 0x00000000 :</output>
```

Comment:
Addresses may display differently on 32-bit and 64-bit systems: perhaps this should use `regex="yes"`.

Reply:
Switched to a regex searching for `0x0+ :` instead of `0x00000000 :`